VAMobile/src/App.tsx

```diff
@@ -268,18 +269,22 @@ export function AuthGuard() {
 
   useEffect(() => {
     console.debug('AuthGuard: initializing')
-    dispatch(initializeAuth())
-
-    const listener = (event: { url: string }): void => {
-      if (event.url?.startsWith('vamobile://login-success?')) {
-        dispatch(handleTokenCallbackUrl(event.url))
+    if (loggedIn && tappedForegroundNotification) {
+      console.debug('User tapped foreground notification. Skipping initializeAuth.')
+      setTappedForegroundNotification(false)
+    } else if (!loggedIn) {
+      dispatch(initializeAuth())
+      const listener = (event: { url: string }): void => {
+        if (event.url?.startsWith('vamobile://login-success?')) {
+          dispatch(handleTokenCallbackUrl(event.url))
+        }
+      }
+      const sub = Linking.addEventListener('url', listener)
+      return (): void => {
+        sub?.remove()
```

Comment:
I found that it was actually causing issue leaving it outside of the block because it was retriggering and causing hang ups on Judy, it is only necessary to have this trigger when logged out